packages/graphql-mini-transforms/src/webpack.ts

```diff
@@ -3,21 +3,37 @@ import {dirname} from 'path';
 import {loader} from 'webpack';
 import {parse, DocumentNode} from 'graphql';
 import {getOptions} from 'loader-utils';
+import validateOptions from 'schema-utils';
 
 import {cleanDocument, extractImports, toSimpleDocument} from './document';
 
 interface Options {
+  generateId?: (normalizedSource: string) => string;
```

Comment:
Instead of this just accepting the source as a string, can we make the argument be an object of `{source: string}`? Right now source is the only argument we need to pass in, but this signature gives us more flexibility to add things like the path, or more information about the concatenated fragments, if that is needed later on.

Reply:
Changed in 13b8012b1b91f340be0491cf5c541dc275d3e349.